src/Sylius/Bundle/PromotionBundle/Form/Type/CatalogPromotionType.php

```diff
@@ -90,6 +87,18 @@ public function buildForm(FormBuilderInterface $builder, array $options): void
             ])
         ;
 
+        $builder->addEventListener(FormEvents::PRE_SET_DATA, function (FormEvent $event): void {
+            /** @var CatalogPromotion $data */
+            $data = $event->getData();
+
+            $form = $event->getForm();
+            $form->add('enabled', CheckboxType::class, [
```

Comment:
I'm unaware of catalog promotion inner workings, but usually enabled/disabled and archived exists in parallel.

Reply:
That't the change to disallow to enable promotion, if it's archived. Archived promotion means for me "cannot be used".